relations/src/shielder/merge.rs

```diff
@@ -0,0 +1,319 @@
+use liminal_ark_relation_macro::snark_relation;
+
+/// It expresses the facts that:
+///  - `first_old_note` is a prefix of the result of hashing together the `token_id`,
```

Comment:
"prefix of ..." is no more - this is just a hash

Reply:
OK, good catch, changed.